src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -1532,12 +1532,10 @@ private MutableBigInteger divideMagnitude(MutableBigInteger div,
         int[] q = quotient.value;
 
 
-        // Must insert leading 0 in rem if its length did not change
-        if (rem.intLen == nlen) {
```

Comment:
Looks like after this change the local nlen variable is used only once, and probably can be removed?

Reply:
BTW the "rem.leftShift(shift);" above may change the "rem.intLen" value, no?

Reply:
Sorry I have looked into the different method=), -> divideLongMagnitude.